web/packages/teleterm/src/ui/services/clusters/clustersService.ts

```diff
@@ -291,9 +296,21 @@ export class ClustersService extends ImmutableStore<types.ClustersServiceState>
     try {
       await this.syncRootCluster(clusterUri);
     } catch (e) {
-      this.notificationsService.notifyError({
+      const cluster = this.findCluster(clusterUri);
+      const clusterName =
+        cluster?.name ||
+        routing.parseClusterUri(clusterUri).params.rootClusterId;
+
+      const notificationId = this.notificationsService.notifyError({
         title: `Could not synchronize cluster ${clusterName}`,
         description: e.message,
+        action: {
+          content: 'Retry',
+          onClick: () => {
+            this.notificationsService.removeNotification(notificationId);
```

Comment:
Yeah, it's a bit confusing, but really cool in cases like this one.
I think it's a JS closure, where an inner function has access to variables from the outer scope (or something like this). So I guess the function is built just before it's invoked?

Reply:
The anonymous function passed as `onClick` for sure is constructed before it's invoked. My issue was that at the time when that function is constructed, `notificationId` doesn't exist yet.

I think I forgot that JS after all is a dynamic language. So that anonymous function has access to the closure above, not just everything instantiated just before the anonymous function is constructed. By the time `onClick` is called, `notificationId` is available within the closure.

This illustrates this even better:

```javascript
function test() {
  let uuid = createNotification(() => {
    console.log(uuid)
    console.log(somethingDefinedAfterCbButWithinClosure)
  })
  let somethingDefinedAfterCbButWithinClosure = 'somethingDefinedAfterCbButWithinClosure'
}

function createNotification(cb) {
  setTimeout(cb, 50)
  return '1234abcd'
}

test()
```
The output of this is:
```
1234abcd
somethingDefinedAfterCbButWithinClosure
```

Reply:
Good example, thanks!